Shazam.php

```diff
@@ -36,6 +33,145 @@ public function __construct()
         // if ($project_id) $this->loadConfig();
     }
 
+
+    /**
+     * Returns all users saved in project config
+     * @param void
+     * @return $js_users : array
+     */
+    public function getJavascriptUsers(){
+        if($this->getSystemSetting('enable-add-user-javascript-permissions')){
+            $js_users = json_decode($this->getProjectSetting(self::JS_USER_KEY));
+            return isset($js_users) ? $js_users : [];
+        }
+    }
+
+
+    /**
+     * Adds $username to project settings, enabling JS editing
+     * @param $username
+     * @return void
+     */
+    public function addJavascriptUser($username) {
+        if(SUPER_USER && $this->getSystemSetting('enable-add-user-javascript-permissions')){ //Check if user has permission to add
+            $existing_users = $this->getJavascriptUsers();
+
+            if(empty($existing_users)){ //if empty simply add to the list
```

Comment:
Seems like this nested if and if isn't necessary...  You already have an array of existing users - why two paths?
```suggestion
            if(! in_array($username, $existing_users)) {
                // Add user to array and save
             } else {
                // user is already in list, don't do anything
             }
```

Reply:
Replaced block with 
```
if(in_array($username, $existing_users)){
                $this->emDebug("Error, username {$username} exists within existing users, update choices");
            } else {
                array_push($existing_users, $username);
                $this->setProjectSetting(self::JS_USER_KEY, json_encode($existing_users));
                $this->sendNotificationEmail($username);
            }
```